pkg/kubectl/cmd/edit.go

```diff
@@ -257,7 +257,12 @@ func RunEdit(f *cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args
 			}
 			err = schema.ValidateBytes(stripComments(edited))
 			if err != nil {
-				return preservedFile(err, file, errOut)
+				results = editResults{
+					file: file,
+				}
+				containsError = true
+				results.header.reasons = append(results.header.reasons, editReason{head: fmt.Sprintf("The edited file fails validation: %v", err)})
```

Comment:
I can do that. But the format of error message isn't perfect, since some fields, such as `path` and `invalid value` are not available directly from the error returned by `ValidateBytes()`. 
eg: 
\# pods "test-pod" was not valid:
\# * : Invalid value: "": couldn't find type: v1.Podddd